pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_drop_column.go

```diff
@@ -349,15 +351,16 @@ func walkDropColumnDependencies(b BuildCtx, col *scpb.Column, fn func(e scpb.Ele
 			}
 		case *scpb.SecondaryIndex:
```

Comment:
I think we still want to check `SecondaryIndexPartial` given that in mixed version and before we fully migrated to `V23_1_SchemaChangerDeprecatedIndexPredicates` we still emit `SecondaryIndexPartial`, see https://github.com/cockroachdb/cockroach/blob/dda41f8bf34e20813a066c59840044530945ded4/pkg/sql/schemachanger/scdecomp/decomp.go#L560-L560